### PR TITLE
a11y adjustments for the Bottom app bar demo

### LINKED ISSUE #16225: Gallery Bottom Action Bar demo needs to be checked for a11y
See https://github.com/flutter/flutter/pull/16196#issuecomment-378656606

### PR DESCRIPTION
**Pull request by DaveShuckerow**
@goderbauer 

fixes #16225

### CODE REVIEW

File: examples/flutter_gallery/lib/demo/material/bottom_app_bar_demo.dart
Comment (goderbauer) [inline examples/flutter_gallery/lib/demo/material/bottom_app_bar_demo.dart]: nit: `explicitChildNodes: false` is default behavior and should probably just be removed.

File: examples/flutter_gallery/lib/demo/material/bottom_app_bar_demo.dart
Comment (goderbauer) [inline examples/flutter_gallery/lib/demo/material/bottom_app_bar_demo.dart]: This is kinda odd. Why should a screen reader user have a different experience of the UI then somebody who doesn't use a screen reader?

**Review by goderbauer — APPROVED** [review]
LGTM, assuming that you have tested this with an actual screen reader.

**Comment by DaveShuckerow** [comment]
That's correct, tested with TalkBack on Android.

I assume that we test Flutter's semantics engine to give the same performance on iOS.  Let me know if that isn't correct.

File: examples/flutter_gallery/lib/demo/material/bottom_app_bar_demo.dart
Comment (DaveShuckerow) [reply examples/flutter_gallery/lib/demo/material/bottom_app_bar_demo.dart]: Since this is a demo, these are dummy buttons, not disabled buttons.  

I decided to tell the screen reader that they are disabled because when I tried it out with a screen reader, I ended up confused.  I partially blindfolded myself and tried tapping those buttons.  When the reader told me I could double-tap to activate them, I did and I was confused when I couldn't find anything changed.

Since the primary purpose of this demo is to showcase normal Flutter code, I am OK with removing those semantics objects.

I'll let you make the final call, since we should apply any solution we come up with to all no-op buttons in the gallery.

File: examples/flutter_gallery/lib/demo/material/bottom_app_bar_demo.dart
Comment (DaveShuckerow) [reply examples/flutter_gallery/lib/demo/material/bottom_app_bar_demo.dart]: I've gone ahead and done that.

I merged this before seeing this comment from mobile.  I'm too used to our internal UI where any comments blocks submitting a change. 

I'll send a PR with this update and remember to check for final comments before merging next time.

**Pull request merged**
